apps/cloud/src/api/mod.rs

```diff
@@ -723,3 +723,207 @@ pub async fn search_workspace(
 
     Json(search_results).into_response()
 }
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use axum:: body::Body;
+    use axum_test_helper::TestClient;
+    use bytes::Bytes;
+    use cloud_database::{CloudDatabase, CreateUser};
+    use futures::{stream, StreamExt};
+    use serde_json::json;
+    use std::sync::Arc;
+    #[tokio::test]
+    async fn test_health_check() {
+        let pool = CloudDatabase::init_pool("sqlite::memory:").await.unwrap();
+        let context = Context {
+            db: pool,
+            ..Context::new().await
+        };
+        let ctx = Arc::new(context);
+        let app = super::make_rest_route(ctx.clone()).layer(Extension(ctx.clone()));
+
+        let client = TestClient::new(app);
+        let resp = client.get("/healthz").send().await;
+        assert_eq!(resp.status(), StatusCode::OK);
+    }
+
+    #[tokio::test]
+    async fn test_query_user() {
+        let pool = CloudDatabase::init_pool("sqlite::memory:").await.unwrap();
+        let context = Context {
+            db: pool,
+            ..Context::new().await
+        };
+        let new_user = context
+            .db
+            .create_user(CreateUser {
+                avatar_url: Some("xxx".to_string()),
+                email: "xxx@xxx.xx".to_string(),
+                name: "xxx".to_string(),
+                password: "xxx".to_string(),
+            })
+            .await
+            .unwrap();
+        let ctx = Arc::new(context);
+        let app = super::make_rest_route(ctx.clone()).layer(Extension(ctx.clone()));
+
+        let client = TestClient::new(app);
+        let url = format!(
+            "/user?email={}&workspace_id=212312",
+            new_user.email
+        );
+        let resp = client.get(&url).send().await;
+        assert_eq!(resp.status(), StatusCode::OK);
+        let resp = client.get("/user").send().await;
+        assert_eq!(resp.status().is_client_error(), true);
+        let resp = client.get("/user?email=fake_email").send().await;
+        assert_eq!(resp.status().is_client_error(), true);
+        let resp = client.get("/user?user_name=fake_parameter").send().await;
+        assert_eq!(resp.status().is_client_error(), true);
+    }
+
+    #[tokio::test]
+    async fn test_make_token() {
```

Comment:
split the test case of correct and wrong requests, such as test_make_token_with_valid_request()/test_make_token_with_invalid_request()